src/iterator.cr

```diff
@@ -99,18 +99,33 @@ module Iterator(T)
     self
   end
 
+  # Returns a new array with the results of running block once for
+  # every element in enum.
+  #
+  # ```
+  # [1, 2, 3].map &.*(2) # => [2, 4, 6]
+  # ```
```

Comment:
This is calling `Array#map`, not `Iterator#map`, so if we actually call `Iterator#map` we get:

```cr
[1, 2, 3].each.map &.*(2) # => Iterator(T)::Map(Array(T)::ItemIterator(Int32), Int32, Int32)(@iter=#<Array(T)::ItemIterator(Int32):0x8652fc0 @array=[1, 2, 3], @index=0>, @func=#<(Int32 -> Int32):0x804b1a0>)
```

A better example might be

```cr
doubled = [1, 2, 3].each.map &.*(2)
doubled.next # => 2
doubled.next # => 4
doubled.next # => 6
```

or even just
```cr
[1, 2, 3].map(&.*(2)).to_a # => [2, 4, 6]
```

The basic oversight is true for all other examples.